src/Contract/Monad.purs

```diff
@@ -0,0 +1,137 @@
+-- |
+-- | All useful functions written in `QueryM` should be lifted into the
+-- | `Contract` monad and available in the same namespace. If anything is
+-- | missing, please contact us.
+newtype Contract (a :: Type) = Contract (QueryM a)
+
+-- Many of these derivations of depending on the underlying `ReaderT` and
+-- asychronous effects,`Aff`.
+derive instance Newtype (Contract a) _
+derive newtype instance Functor Contract
+derive newtype instance Apply Contract
+derive newtype instance Applicative Contract
+derive newtype instance Alt Contract
+derive newtype instance Plus Contract
+derive newtype instance Bind Contract
+derive newtype instance Monad Contract
+derive newtype instance MonadEffect Contract
+derive newtype instance MonadAff Contract
+derive newtype instance Semigroup a => Semigroup (Contract a)
+derive newtype instance Monoid a => Monoid (Contract a)
+-- Utilise JavaScript's native `Error` via underlying `Aff` for flexibility:
+derive newtype instance MonadThrow Error Contract
+derive newtype instance MonadError Error Contract
+derive newtype instance MonadRec Contract
+
+instance MonadAsk ContractConfig Contract where
+  -- Use the underlying `ask`:
+  ask = Contract $ ContractConfig <$> ask
+
+instance MonadReader ContractConfig Contract where
+  -- Use the underlying `local` after dimapping and unwrapping:
+  local f contract = Contract $ local (dimap wrap unwrap f) (unwrap contract)
+
+-- | The config for `Contract` is just a newtype wrapper over the underlying
+-- | `QueryM` config.
+newtype ContractConfig = ContractConfig QueryConfig
+
+derive instance Newtype ContractConfig _
+
+-- | Throws an `Error` for any showable error using `Effect.Exception.throw`
+-- | and lifting into the `Contract` monad.
+throwContractError :: forall (e :: Type) (a :: Type). Show e => e -> Contract a
+throwContractError = liftEffect <<< throw <<< show
+
+-- | Runs the contract, essentially `runReaderT` but with arguments flipped.
+runContract :: forall (a :: Type). ContractConfig -> Contract a -> Aff a
+runContract config = flip runReaderT (unwrap config) <<< unwrap
+
+-- | Same as `runContract` discarding output.
+runContract_ :: forall (a :: Type). ContractConfig -> Contract a -> Aff Unit
+runContract_ config = void <<< flip runReaderT (unwrap config) <<< unwrap
+
+-- | Creates a default `ContractConfig` with a Nami wallet inside `Aff` as
+-- | required by the websockets.
+defaultContractConfigAff :: Aff ContractConfig
+defaultContractConfigAff = do
+  wallet <- Just <$> mkNamiWalletAff
+  ogmiosWs <- QueryM.mkOgmiosWebSocketAff QueryM.defaultOgmiosWsConfig
+  datumCacheWs <-
+    QueryM.mkDatumCacheWebSocketAff QueryM.defaultDatumCacheWsConfig
+  usedTxOuts <- newUsedTxOuts
+  pure $ ContractConfig
+    { ogmiosWs
+    , datumCacheWs
+    , wallet
+    , serverConfig: QueryM.defaultServerConfig
+    , usedTxOuts
+    }
+
+-- | Same as `defaultContractConfigAff` but lifted into `Contract`.
+defaultContractConfig :: Contract ContractConfig
```

Comment:
I'm not sure how useful this will be, since we need a `ContractConfig` to run any `Contract` action. But I suppose we can keep it (maybe it would be useful with `local` or something?)

Reply:
It could be useful for local, although I wonder if `defaultContractConfig` is a better name for the `Aff` version since that's what everyone is using. Then call the `Contract` version `defaultContractConfigLifted` perhaps.

Reply:
I think that's the best/clearest way to name them